app/tests/spec/lib/crypto/recovery-keys.js

```diff
@@ -0,0 +1,145 @@
+          });
+      });
+
+      it('should have correct lenght', () => {
+        return RecoveryKeys.generateRecoveryKey(1000)
+          .then((key) => {
+            assert.equal(key.length, 1000);
+          });
+      });
+    });
+
+    describe('getRecoveryEncryption', () => {
+      it('throws if no uid', () => {
+        return RecoveryKeys.getRecoveryEncryption(undefined, recoveryKey)
+          .then(assert.fail, (err) => {
+            assert.equal(err.message, 'Missing uid');
+          });
+      });
+
+      it('throws if no recoveryKey', () => {
+        return RecoveryKeys.getRecoveryEncryption(uid, undefined)
+          .then(assert.fail, (err) => {
+            assert.equal(err.message, 'Missing recoveryKey');
+          });
+      });
+
+      it('should create recoveryKeyId and recoveryEnc', () => {
+        return RecoveryKeys.getRecoveryEncryption(uid, recoveryKey)
+          .then((encryption) => {
+            assert.equal(validate.isHexValid(encryption.recoveryEnc), true, 'is hex value');
+            assert.equal(validate.isHexValid(encryption.recoveryKeyId), true, 'is hex value');
+            assert.equal(encryption.recoveryEnc, expectedRecoveryEnc, 'recoveryEnc is correct');
+            assert.equal(encryption.recoveryKeyId, expectedRecoveryKeyId, 'recoveryKeyId is correct');
+          });
+      });
+    });
+
+    describe('createRecoveryBundle', () => {
+      let encryption;
+
+      beforeEach(() => {
+        return RecoveryKeys.getRecoveryEncryption(uid, recoveryKey)
+          .then((result) => encryption = result);
+      });
+
+      it('throws if no recoveryKeyId', () => {
+        return RecoveryKeys.createRecoveryBundle(undefined, encryption.recoveryEnc)
+          .then(assert.fail, (err) => {
+            assert.equal(err.message, 'Missing recoveryKeyId');
+          });
+      });
+
+      it('throws if no recoveryEnc', () => {
+        return RecoveryKeys.createRecoveryBundle(encryption.recoveryKeyId, undefined)
+          .then(assert.fail, (err) => {
+            assert.equal(err.message, 'Missing recoveryEnc');
+          });
+      });
+
+      it('should create recovery bundle', () => {
+        return RecoveryKeys.createRecoveryBundle(encryption.recoveryKeyId, encryption.recoveryEnc, recoveryData, /* encryptionOptions */)
+          .then((bundle) => {
+            assert.ok(bundle, 'bundle exists');
+            // assert.equal(bundle, expectedBundle, 'bundle matches');
+          });
+      });
+    });
+
+    describe('unbundleRecoveryData', () => {
+      let encryption, encryptedRecoveryData;
+
+      beforeEach(() => {
+        return RecoveryKeys.getRecoveryEncryption(uid, recoveryKey)
+          .then((result) => encryption = result)
+          .then(() => RecoveryKeys.createRecoveryBundle(encryption.recoveryKeyId, encryption.recoveryEnc, recoveryData))
+          .then((result) => encryptedRecoveryData = result);
+      });
+
+      it('throws if no recoveryEnc', () => {
```

Comment:
TBH, I'm not sure there's a lot of value to test for missing required parameters, seems like these tests obscure the more important stuff.